appmetrics/database.go

```diff
@@ -0,0 +1,159 @@
+package appmetrics
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+
+	"strings"
+
+	"github.com/xeipuuv/gojsonschema"
+)
+
+type AppMetricEventType string
+
+// Value is `json.RawMessage` so we can send any json shape, including strings
+// Validation is handled using JSON schemas defined in validators.go, instead of Golang structs
+type AppMetric struct {
+	Event      AppMetricEventType `json:"event"`
+	Value      json.RawMessage    `json:"value"`
+	AppVersion string             `json:"app_version"`
+	OS         string             `json:"os"`
+}
+
+type AppMetricValidationError struct {
+	Metric AppMetric
+	Errors []gojsonschema.ResultError
+}
+
+const (
+	// Events for testing the system
+	TestEvent1 AppMetricEventType = "go/test1"
```

Comment:
It's a bit odd to have testing code not in `_test.go` file, can we avoid it and move it to the `_test.go` file?
If they are not used other than in test, is there value in testing with them? I'd probably test using the `NavigateToCofx` event type or use a schema specifically for testing (you'd have to pass it to `Validate` function in this case)
@Samyoul what's idiomatic in this cases?

Reply:
There is no value in custom events. I added them in the first place because I wanted to test JSON Schema.